proxy/http2/Http2Stream.h

```diff
@@ -164,17 +196,22 @@ class Http2Stream : public ProxyTransaction
   Http2StreamState _state = Http2StreamState::HTTP2_STREAM_STATE_IDLE;
   int64_t _http_sm_id     = -1;
 
-  HTTPHdr _req_header;
-  MIOBuffer _request_buffer = CLIENT_CONNECTION_FIRST_READ_BUFFER_SIZE_INDEX;
-  int64_t read_vio_nbytes;
+  HTTPHdr _recv_header;
+  MIOBuffer _recv_buffer = CLIENT_CONNECTION_FIRST_READ_BUFFER_SIZE_INDEX;
   VIO read_vio;
   VIO write_vio;
 
   History<HISTORY_DEFAULT_SIZE> _history;
   Milestones<Http2StreamMilestone, static_cast<size_t>(Http2StreamMilestone::LAST_ENTRY)> _milestones;
 
-  bool trailing_header = false;
-  bool has_body        = false;
+  bool _possible_trailing_header = false;
+  bool _expect_send_trailer      = false;
+  bool _expect_receive_trailer   = false;
+
+  bool has_body = false;
+
+  bool _outbound_flag     = false;
```

Comment:
This name is confusing a little, because a stream can be initiated by a server on an inbound connection (e.g. Server Push). Since a stream (ProxyTransaction) knows its owner session, we should be able to ask the session whether this stream is on an inbound session or an outbound session.

I'd remove this variable, but please rename it accordingly if you want to keep it.

Reply:
Yes, good observation. Thank you. I updated the function to query the session whether it is outbound or not.